src/main/kotlin/io/github/mojira/arisa/modules/CrashInfoModule.kt

```diff
@@ -0,0 +1,92 @@
+package io.github.mojira.arisa.modules
+
+import arrow.core.Either
+import arrow.core.extensions.fx
+import arrow.core.left
+import arrow.core.right
+import arrow.syntax.function.partially2
+import io.github.mojira.arisa.domain.Issue
+import io.github.mojira.arisa.infrastructure.AttachmentUtils
+import com.urielsalis.mccrashlib.Crash
+import com.urielsalis.mccrashlib.CrashReader
+import com.urielsalis.mccrashlib.deobfuscator.getSafeChildPath
+import java.nio.file.Files
+import java.time.Instant
+
+class CrashInfoModule(
+    private val crashReportExtensions: List<String>,
+    private val crashReader: CrashReader
+) : Module {
+    override fun invoke(issue: Issue, lastRun: Instant): Either<ModuleError, ModuleResponse> = with(issue) {
+        Either.fx {
+            val crashes = AttachmentUtils(crashReportExtensions, crashReader).extractCrashesFromAttachments(issue)
+            val newCrashes = assertContainsNewCrash(crashes, lastRun).bind()
+            uploadDeobfuscatedCrashes(issue, newCrashes)
+
+            newCrashes
+                .filter { it.second is Crash.Minecraft }
+                .filterNot { it.first.name.endsWith("deobfuscated.txt") }
+                .forEach {
+                    if (!description!!.contains("""\{code.*${it.first.name}]}(\S|\s)*\{code}""".toRegex())) {
```

Comment:
Is `(\S|\s)` to match line breaks? Maybe the complete `(\S|\s)*\{code}` part can be removed, it is rather unlikely that the description contains `{code:title=...}` which is not actually a code block.

Additionally we should probably only add the crash report to the description if there is none, to prevent adding multiple crash report snippets. Therefore we should probably only consider `newCrashes` if its size is 1, otherwise Arisa might pick the 'wrong' crash report in case a user uploads unrelated ones.

The regex could probably changed to this:
```suggestion
                    // Check if description already contains code block which might contain crash report
                    // Note: Jira seems to allow spaces between most of the characters
                    if (!description!!.contains("""\{code:\s*title\s*=.*\[\^.+\.txt\s*]\s*}""".toRegex())) {
```

Reply:
Maybe this regex is too unspecific though and causes some false negatives, e.g. when `{code}` blocks link to a non-crash attachment (e.g. code snippet file when a code analysis was done). But I assume that this will happen rarely, so it might be fine for now and we can adjust it later if necessary.